include/dxc/DxilRootSignature/DxilRootSignature.h

```diff
@@ -388,6 +385,30 @@ bool VerifyRootSignature(_In_ const DxilVersionedRootSignatureDesc *pDesc,
                          _In_ llvm::raw_ostream &DiagStream,
                          _In_ bool bAllowReservedRegisterSpace);
 
+class ScopedVersionedRootSignature {
+  const DxilVersionedRootSignatureDesc *m_pRootSignature;
+
+public:
+  ScopedVersionedRootSignature() : m_pRootSignature(nullptr) {}
+  ScopedVersionedRootSignature(
```

Comment:
Could have a constructor that takes `ScopedVersionedRootSignature&&` to move ownership, and delete ones that allow copy, since this should act more like `unique_ptr`.

If interfaces that init/modify root sig take the object instead of the ptr, then access could be further protected.

Reply:
Good point about approaching unique_ptr semantics